<!--
  ~  Licensed to the Apache Software Foundation (ASF) under one
  ~  or more contributor license agreements.  See the NOTICE file
  ~  distributed with this work for additional information
  ~  regarding copyright ownership.  The ASF licenses this file
  ~  to you under the Apache License, Version 2.0 (the
  ~  "License"); you may not use this file except in compliance
  ~  with the License.  You may obtain a copy of the License at
  ~
  ~    http://www.apache.org/licenses/LICENSE-2.0
  ~
  ~  Unless required by applicable law or agreed to in writing,
  ~  software distributed under the License is distributed on an
  ~  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
  ~  KIND, either express or implied.  See the License for the
  ~  specific language governing permissions and limitations
  ~  under the License.
  ~
  -->
<test-suite
             xmlns="urn:xml.testframework.asterix.apache.org"
             ResultOffsetPath="results"
             QueryOffsetPath="queries_sqlpp"
             QueryFileExtension=".sqlpp">
  <test-group name="external-library-python">
    <test-case FilePath="external-library">
      <compilation-unit name="mysentiment">
        <output-dir compare="Text">mysentiment</output-dir>
      </compilation-unit>
    </test-case>
    <test-case FilePath="external-library">
      <compilation-unit name="python-fn-escape">
        <output-dir compare="Text">python-fn-escape</output-dir>
        <expected-error>ImportError: Module was not found in library</expected-error>
      </compilation-unit>
    </test-case>
    <test-case FilePath="external-library">
      <compilation-unit name="py_nested_access">
        <output-dir compare="Clean-JSON">py_nested_access</output-dir>
      </compilation-unit>
    </test-case>
    <test-case FilePath="external-library" check-warnings="true">
      <compilation-unit name="py_function_error">
        <output-dir compare="Clean-JSON">py_function_error</output-dir>
        <expected-warn>ASX0201: External UDF returned exception. Returned exception was: ArithmeticError: oof</expected-warn>
      </compilation-unit>
    </test-case>
  </test-group>
</test-suite>
